include/async/mutex.hpp

```diff
@@ -309,8 +309,8 @@ namespace detail {
 				}else{
 					assert(!shared_cnt_);
 					st_ = state::shared;
-					shared_cnt_ = 1;
-					while(waiters_.front()->desired == state::shared) {
+					shared_cnt_ = 0;
```

Comment:
The `shared_cnt_ = 0;` statement can be dropped entirely since we assert that `!shared_cnt_` two lines above.